components/ChipTextField/examples/MDCChipTextField.m

```diff
@@ -50,6 +51,54 @@ - (instancetype)initWithFrame:(CGRect)frame {
   return self;
 }
 
+#pragma mark - Property Getter
+
+- (NSArray<MDCChipView *> *)chipViews {
+  return [self.mutableChipViews copy];
+}
+
+#pragma mark - Public API
+
+- (void)appendChipWithText:(NSString *)text {
+  MDCChipView *chipView = [[MDCChipView alloc] init];
+  chipView.titleLabel.text = text;
+  chipView.translatesAutoresizingMaskIntoConstraints = NO;
+
+  [self appendChipView:chipView];
+}
+
+- (void)appendChipView:(MDCChipView *)chipView {
+  if ([self.chipTextFieldDelegate respondsToSelector:@selector(chipField:shouldAddChip:)]) {
+    NSInteger indexToAppend = self.mutableChipViews.count;
+    BOOL shouldAppend = [self.chipTextFieldDelegate chipTextField:self
+                                                shouldAddChipView:chipView
+                                                          atIndex:indexToAppend];
+    if (!shouldAppend) {
+      return;
+    }
+  }
+
+  [self.chipsContainerView appendChipView:chipView];
+
+  // recalculate the layout to get a correct chip frame values
+  [self.chipsContainerView layoutIfNeeded];
+  [self.mutableChipViews addObject:chipView];
+
+  [self clearChipsContainerOffsetWithConstant:kTextToEnterPlaceholderLength];
```

Comment:
What is this constant (kTextToEnterPlaceholderLength) for?

Reply:
It is a constant number currently used to fix the weird typing offset issue.